### PR TITLE
add visual validation util classes and methods to esb tooling

### PR DESCRIPTION
**Pull request by rosensilva**

### CODE REVIEW

**Comment by dakshika** [comment]
Seems we need to replace the warning icon with error icon.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/edit/parts/CallMediatorEditPart.java [hunk 1]
```diff
@@ -336,6 +347,40 @@ public IFigure getToolTip() {
...
+            if (model.getElement() instanceof CallMediatorImpl) {
+                CallMediatorImpl callMediatorDataModel = (CallMediatorImpl) model.getElement();
+                try {
+                    org.apache.synapse.mediators.builtin.CallMediator callMediator;
+                    callMediator = CallMediatorTransformer.createCallMediator((EsbNode) callMediatorDataModel, true);
+                    CallMediatorSerializer callMediatorSerializer = new CallMediatorSerializer();
+                    OMElement omElement = callMediatorSerializer.serializeSpecificMediator(callMediator);
+                    // This will add a marker in the visual view if the data model is incomplete
+                    if (StringUtils
+                            .isEmpty(MediatorValidationUtil.validateMediatorsFromOEMElement(omElement, "call"))) {
```
Comment (abeykoon) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/edit/parts/CallMediatorEditPart.java]: Shall we move "call" to a constant?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CallMediatorTransformer.java [hunk 2]
```diff
@@ -117,17 +120,26 @@ public void transformWithinSequence(TransformationInfo information, EsbNode subj
...
-        for (int i = 0; i < nameSpacedProperty.getNamespaces().keySet().size(); ++i) {
-            String prefix = (String) nameSpacedProperty.getNamespaces().keySet().toArray()[i];
-            String namespaceUri = nameSpacedProperty.getNamespaces().get(prefix);
-            synapseXPath.addNamespace(prefix, namespaceUri);
+    public static ResolvingEndpoint getEndpointFromXpath(NamespacedProperty nameSpacedProperty, boolean isForValidation)
+            throws JaxenException {
+        SynapseXPath synapseXPath;
+        ResolvingEndpoint resolvingEndpoint;
+        if (StringUtils.isEmpty(nameSpacedProperty.getPropertyValue()) && !isForValidation) {
+            synapseXPath = new SynapseXPath("FACTORY/DEFAULT/VALUE");
```
Comment (abeykoon) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CallMediatorTransformer.java]: Let's move this to a meaningful constant visible across all mediator implementations.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/edit/parts/CallMediatorEditPart.java (hunk 1 above)
Comment (rosensilva) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/edit/parts/CallMediatorEditPart.java]: Fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CallMediatorTransformer.java (hunk 2 above)
Comment (rosensilva) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CallMediatorTransformer.java]: Fixed

**Comment by rosensilva** [comment]
> Seems we need to replace the warning icon with error icon.

added new error-symbol

**Pull request merged**